pkg/client/image.go

```diff
@@ -144,17 +156,60 @@ func (c *client) ImagePush(ctx context.Context, imageName string, opts *ImagePus
 	return result, nil
 }
 
-func (c *client) ImageDelete(ctx context.Context, imageName string) (*apiv1.Image, error) {
+func (c *client) ImageDelete(ctx context.Context, imageName string, opts *ImageDeleteOptions) (*apiv1.Image, error) {
 	image, err := c.ImageGet(ctx, imageName)
 	if apierrors.IsNotFound(err) {
 		return nil, nil
 	} else if err != nil {
 		return nil, err
 	}
+	if err != nil {
+		return nil, err
+	}
+	if includesTag(imageName) && len(image.Tags) != 1 {
+		for _, ref := range image.References {
+			if ref == imageName {
+				imgToDel := &apiv1.Image{
+					TypeMeta:    image.TypeMeta,
+					ObjectMeta:  image.ObjectMeta,
+					Digest:      image.Digest,
+					Repository:  image.Repository,
+					Tags:        []string{strings.Split(imageName, ":")[0]},
+					References:  []string{ref},
+					ClusterName: "",
+				}
+				imgToDel.ResourceVersion = "del"
```

Comment:
what is this doing ? seems odd to set ResourceVersion to "del"

Reply:
Don't abuse resourceVersion like this. We want to make sure our API is behaving like a proper k8s api. This special logic wouldn't be good for that. For example, Darren mentioned today that the next phase of refactor would be migrating this api to a Custom Resource Definition. When we do that, you won't be able to manipulate resourceVersion like this.

I suspect that much of this logic will change given the refactor we discussed this morning. But at the end of the day, if the client wants to untag, you should `GET` the image, then modify the list of tags in that object, then do a `PUT` with that image object.

Server side, you will have to basically figured out what has changed - get all the tags we've stored server-side and compare them to the incoming tags.  This could mean that in one PUT, you are adding and deleting multiple tags. Whatever the client PUTs, you have to make reality server side. So, the strategy will have to do a bit of slice comparison logic.

Reply:
And if this is a true delete of the image, you should be doing a `DELETE` via the client